### DISCUSSION TITLE
Need for a rust or zig sdk?

### DISCUSSION THREAD

**Discussion by davidon-top** [zitadel/zitadel#6383] [Contributors]
Hello there,

After getting familiar with zitadel i kinda want to make a sdk that would help me and (hopefully) many other developers with using zitadel, but i don't know if i should make it for rust or zig (or potentially ocaml tho i don't have much experience with that language yet). I've personally been enjoying zig more and more every day but i understand that rust is more popular.

If someone from the zitadel team could share their thoughts on this or get in touch with me that would be amazing.

**Comment by mffap** [comment]
Hi @davidon-top have you checked out https://github.com/smartive/zitadel-rust for rust? 
It's not maintained by zitadel itself, but a close partner of ours.

Haven't tried zig, but grpc allows to autogenerate your own sdk/client, as mentioned here: https://zitadel.com/docs/examples/sdks#missing-an-sdk

I found the following lib to maybe help you along
https://github.com/Arwalk/zig-protobuf

**Reply by davidon-top** [reply]
i only looked at the docs because i thought sdks would be there, sorry.
